File: Algorithms/942.DI-String-Match/i_string_match.go
func diStringMatch(S string) []int {
    res := []int{}
    start := 0
    end := len(S)
    for _, ele := range S {
        if string(ele) == "I" {
            res = append(res, start)
            start += 1
        } else {
            res = append(res, end)
            end -= 1
        }
    }
    res = append(res, start)
    return res
}

File: Algorithms/942.DI-String-Match/i_string_match.py
class Solution:
    def diStringMatch(self, S: str) -> List[int]:
        res = []
        length = len(S)
        start = 0
        end = length
        for i in range(length):
            if S[i] == "I":
                res.append(start)
                start += 1
            else:
                res.append(end)
                end -= 1
        res.append(start)
        return res
